src/plugins/ethereum/ethAddress.js

```diff
@@ -0,0 +1,39 @@
+// @flow
+
+import {isAddress, toChecksumAddress} from "web3-utils";
+
+import * as C from "../../util/combo";
+import {compatibleParser} from "../../util/compat";
+
+export opaque type EthAddress: string = string;
+
+export function parseAddress(s: string): EthAddress {
```

Comment:
This function has a contract. Could you document it? Specifically:

  - The input string should be a 40-byte hex digest of an Ethereum
    address, with optional `0x` prefix. (Is that `0x` case-sensitive?)
  - It may be all-lowercase or all-uppercase.
  - It may also be mixed-case, in which case it MUST match the checksum.

Maybe this is obvious to you, but Eth noobs like me have to decipher it
from the tests. It’s a testament to the tests that I can do this! But it
should also be a testament to the docs that I don’t have to. ;-)